lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -210,6 +210,18 @@ namespace Js
                     return nullptr;
                 }
 
+                // Ignore special properties (ex: Array.length)
```

Comment:
Could you move this down after the last loop?
We only need to mark the special properties if there is something else to enumerate (ie non-null prototype).

Basically, if all the following checks haven't returned null, then we will proceed to enumerate the prototype.